src/ophyd_async/core/_table.py

```diff
@@ -102,7 +102,10 @@ def numpy_table(self, selection: slice | None = None) -> np.ndarray:
             if array is None:
                 array = np.empty(v.shape, dtype=self.numpy_dtype())
             array[k] = v
-        assert array is not None
+        if array is None:
+            # TODO: discuss whether this code is actually reachable
```

Comment:
It is reachable, when there are no columns in the table, so good idea giving it an error message!

Reply:
I removed the TODO